### DISCUSSION TITLE
Koda app feels slower than it used to be

### DISCUSSION THREAD

**Discussion by prury** [kodadot/nft-gallery#10294] [General]
just general feelings: 

Navigating trough pages seems to be slower than normal
Images take more time to load
many refreshes required in order for things to appear

what can be done?

**Comment by vikiival** [comment]
cc @kodadot/internal-dev

**Comment by preschian** [comment]
> Images take more time to load

Oh, I guess this is the image endpoint from fxart? If yes, I just updated the headers yesterday so that the browser can cache it [kodadot/private-workers@`478f24f` (#153)](https://github.com/kodadot/private-workers/pull/153/commits/478f24fe021aa8a7bcd5773bb46d8916fcb3d595). Previously I forgot to add that, should be faster now. Here are the images from the browser cache:

<img width="800" alt="Screenshot 2024-05-15 at 10 13 14" src="https://github.com/kodadot/nft-gallery/assets/734428/acd59f90-3326-4bdc-bba8-ba33c8b40235">


For non-images from image workers, etag has not yet been utilised. I will improve that to make it faster https://github.com/kodadot/workers/blob/eda0b2477a27126022c6fd142fc5dfdfa379b0c1/image/src/routes/ipfs.ts#L74

> Navigating trough pages seems to be slower than normal

On Canary, navigating to the user page is noticeably slow, probably due to some blocking requests for the profile endpoints.

---

Anyway, providing us with specific endpoints or pages would be very helpful for checking.

**Reply by vikiival** [reply]
> probably due to some blocking requests for the profile endpoints.

Cc @daiagi

**Reply by daiagi** [reply]
* #10302

**Comment by Jarsen136** [comment]
It could optimise the loading time for the drop minting page, by cutting some unnecessary requests. https://github.com/kodadot/nft-gallery/pull/10345
